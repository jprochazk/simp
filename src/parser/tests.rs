fn parse(code: &str) -> String {
    match super::parse(code) {
        Ok(ast) => format!("{ast}"),
        Err(err) => format!("{}", err.render(code)),
    }
}

#[glob_test::glob("./inputs/**/*.simp")]
fn test_parse(path: &std::path::Path) {
    let input = std::fs::read_to_string(path).unwrap();
    insta::assert_snapshot!(parse(&input));
}
