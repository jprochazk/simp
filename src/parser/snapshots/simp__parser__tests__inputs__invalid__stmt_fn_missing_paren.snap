---
source: src/parser/tests.rs
expression: parse(&input)
---
expected TOK_LPAREN, found TOK_LBRACE
1 |  fn f {}
  |       ^
